MonoGame.Framework/Graphics/Texture2D.cs

```diff
@@ -306,7 +290,32 @@ private static void ConvertToABGR(int pixelHeight, int pixelWidth, int[] pixels)
                 uint pixel = (uint)pixels[i];
                 pixels[i] = (int)((pixel & 0xFF00FF00) | ((pixel & 0x00FF0000) >> 16) | ((pixel & 0x000000FF) << 16));
             }
+        }
+
+        private void ValidateParams<T>(int level, int arraySlice, Rectangle? rect, T[] data, int startIndex, int elementCount) where T : struct
+        {
+            var textureBounds = new Rectangle(0, 0, Math.Max(width >> level, 1), Math.Max(height >> level, 1));
+            var checkedRect = rect.HasValue ? rect.Value : textureBounds;
+            if (level < 0 || level >= LevelCount)
+                throw new ArgumentException("level must be smaller than the number of levels in this texture.");
+            if (arraySlice > 0 && !GraphicsDevice.GraphicsCapabilities.SupportsTextureArrays)
+                throw new ArgumentException("Texture arrays are not supported on this graphics device", "arraySlice");
+            if (arraySlice < 0 || arraySlice >= ArraySize)
+                throw new ArgumentException("arraySlice must be smaller than the ArraySize of this texture and larger than 0.");
+            if (!textureBounds.Contains(checkedRect))
+                throw new ArgumentException("Rectangle must be inside the texture bounds", "rect");
+            if (data == null)
+                throw new ArgumentNullException("data");
+            var tSize = Marshal.SizeOf(typeof(T));
+            var fSize = Format.GetSize();
+            if (tSize > fSize || fSize % tSize != 0)

```

Comment:
No, this is supposed to be like that. 
XNA requires tSize <= fSize AND fSize % tSize == 0. Intuitively this means that x structs must fit one pixel where x is a whole number. If we negate the conditions we get !(tSize <= fSize) OR !(fSize % tSize == 0) which simplifies to the check here. 

E.g. If your format is Color fSize is 4. Some valid T's are byte, byte2 and Color. Invalid T's are byte3 and anything larger than 4 bytes because you can't put *one* pixel in these types.